diff --git a/package.json b/package.json
index e97de4d..60fbc03 100644
--- a/package.json
+++ b/package.json
@@ -37,7 +37,6 @@
     "jest-fetch-mock": "^1.5.0",
     "jest-serializer-vue": "^1.0.0",
     "node-sass": "^4.9.1",
-    "puppeteer": "^1.5.0",
     "regenerator-runtime": "^0.11.1",
     "sass-loader": "^7.0.3",
     "uglifyjs-webpack-plugin": "^1.2.5",
@@ -48,7 +47,7 @@
     "vue-template-compiler": "^2.5.16",
     "webpack": "^4.7.0",
     "webpack-bundle-analyzer": "^3.3.2",
-    "webpack-cli": "^2.1.2",
+    "webpack-cli": "^3.1.1",
     "webpack-dev-server": "^3.1.4"
   },
   "jest": {
diff --git a/src/components/CurlCodeTab.vue b/src/components/CurlCodeTab.vue
index 2fa5cea..9fba4a3 100644
--- a/src/components/CurlCodeTab.vue
+++ b/src/components/CurlCodeTab.vue
@@ -21,6 +21,11 @@
                 class="md-primary"
                 @click="$emit('copy-output-code', computedCurlCode)"
             >copy</md-button>
+            <md-button
+                class="md-primary"
+                @click="executeCurlRequest"
+            >Execute</md-button>
+            <codemirror :value="curlResponse" :options="editorOptions"></codemirror>
         </md-content>
     </div>
 </template>
@@ -29,7 +34,7 @@
 
 require('codemirror/mode/shell/shell');
 import get from 'lodash/get';
-import { mapGetters } from 'vuex';
+import { mapGetters, mapActions } from 'vuex';
 
 export default {
     name: 'CurlCodeTab',
@@ -48,7 +53,7 @@ export default {
         };
     },
     computed: {
-        ...mapGetters(['inputData']),
+        ...mapGetters(['inputData','curlResponse']),
         computedHeadersStr: function () {
             const headers = Object.keys(this.inputData.requestHeaders);
             if (headers.length > 0) {
@@ -82,6 +87,9 @@ export default {
             `${this.computedHeadersStr}${this.computedRequestBodyStr}"`+
             `${this.inputData.fetchUrl}"${ this.computedCopyToClipStr }`;
         }
+    },
+    methods:{
+        ...mapActions(['executeCurlRequest'])
     }
 };
 </script>
diff --git a/src/store/async-actions/fetch-process-request.js b/src/store/async-actions/fetch-process-request.js
index c4f94c9..42dd528 100644
--- a/src/store/async-actions/fetch-process-request.js
+++ b/src/store/async-actions/fetch-process-request.js
@@ -54,4 +54,9 @@ export function fetchResponse({ getters, dispatch }) {
         });
 }
 
+export function executeCurlRequest({ getters, dispatch }) {
+    console.log("here");
+    dispatch('setCurlResponse', 'My curl resp');
+}
+
 export default fetchResponse;
\ No newline at end of file
diff --git a/src/store/index.js b/src/store/index.js
index f0f439b..107101c 100644
--- a/src/store/index.js
+++ b/src/store/index.js
@@ -1,7 +1,8 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
 import {
-    fetchResponse
+    fetchResponse,
+    executeCurlRequest
 } from './async-actions/fetch-process-request';
 import {
     getSavedRequest
@@ -24,7 +25,9 @@ export const store = new Vuex.Store({
         snackBar: {
             visible: false,
             text: ''
-        }
+        },
+        curlCode: "",
+        curlResponse: ""
     },
     getters: {
         inputData: state => state.inputData,
@@ -34,7 +37,9 @@ export const store = new Vuex.Store({
         requestHeaders: state => state.inputData.requestHeaders,
         response: state => state.response,
         snackBarState: state => state.snackBar.visible,
-        snackBarText: state => state.snackBar.text
+        snackBarText: state => state.snackBar.text,
+        curlCode: state => state.curlCode,
+        curlResponse: state => state.curlResponse
     },
     mutations: {
         setInputData(state, inputData){
@@ -73,7 +78,13 @@ export const store = new Vuex.Store({
         },
         hideSnackBar(state){
             state.snackBar.visible = false;
-        }
+        },
+        setCurlCode(state, curlCode){
+            state.curlCode = curlCode;
+        },
+        setCurlResponse(state, requestBody){
+            state.curlResponse = requestBody;
+        },
     },
     actions: {
         setInputData({commit}, inputData){
@@ -103,6 +114,10 @@ export const store = new Vuex.Store({
         hideSnackBar({commit}){
             commit('hideSnackBar');
         },
+        setCurlResponse({commit}, requestBody){
+            commit('setCurlResponse', requestBody);
+        },
+        executeCurlRequest,
         fetchResponse,
         getSavedRequest,
         saveRequest
